### DISCUSSION TITLE
How to detect SSE connection lost in dart

### DISCUSSION THREAD

**Discussion by ghost** [pocketbase/pocketbase#4639] [Q&A]
Is it possible to detect SSE disconnect from frontend flutter, I'm seeing some users are no longer getting realtime updates as our app is heavily dependent on realtime data I need to detect it and force reload the business logic. Is it possible to catch with try catch if so do I use it inside the callback?

**Comment by ganigeorgiev** [comment]
We currently don't expose the underlying SSE client of the [RealtimeService](https://pub.dev/documentation/pocketbase/latest/pocketbase/RealtimeService-class.html).

> I'm seeing some users are no longer getting realtime updates...

How did you confirm this? If you are relying on the logs, are you sure that you are not mistaking it with the 5min idle disconnect->reconnect attempts?

Note that By default in case of an idle SSE connection or SSE connection interruption we'll try to repeatedly reconnect in the background until it succeed (_as long as the app is running_) or all subscriptions are cancelled, and usually users shouldn't worry about disconnects/reconnects because it is handled automatically.

I could consider adding a stream to allow listening to SSE disconnects but it is ambiguous when that should fire - when unsubscribing from all subscriptions, on temp connection interruption, on every failed reconnect attempt, etc.

But again, ideally users shouldn't deal with this manually. If you want to check whether an SDK instance has active realtime connection you can use the `pb.realtime.clientId` field as a flag (empty string means no active realtime client).

Additionally, this is currently only vaguely mentioned in the readme but if you want to execute something _after_ connect/reconnect, you can attach to the generic realtime `PB_CONNECT` topic:

```dart
pb.realtime.subscribe('PB_CONNECT', callback);
```

**Reply by ghost** [reply]
> How did you confirm this? If you are relying on the logs, are you sure that you are not mistaking it with the 5min idle disconnect->reconnect attempts?

It cannot be idle for 5 minutes since when a person signs in, the record is subscribed to and updated every minute. 

It happens on a regular basis; if the boolean in the record is set to false, they should leave the room as specified in the callback on update event, but the user simply hangs with no update from the server, so I'd like to know if the user is not receiving any SSE from the server and resubscribe to the topic or manually remove the user from the room. Should I create a cron/timer to verify whether 'pb.realtime.clientId' is empty and remove the user from the room?

Also is there limitation with SSE, I heard ~100 but my workload is around 200 user (not topic), can I increase the limit.

**Reply by ganigeorgiev** [reply]
> but the user simply hangs with no update from the server

This doesn't sound right and I suspect error on the client-side but I also lack context about the full picture of your use case and without a _minimal_ reproducible example I'm not sure what to recommend or help.

> Also is there limitation with SSE, I heard ~100 but my workload is around 200 user (not topic), can I increase the limit.

The 100 limit you are referring to is probably the HTTP2 max simultaneous open connections but this is for a single client, not the max persistent connections/users the PocketBase server can handle, aka. you can have 100k connected clients each with established max ~100 persistent connection to the PocketBase server (_which if you use the Dart SDK never actually happens because the SDK tries to maintain a single connection per instance; additionally note that this max connections limit is usually only for web, aka. browsers_).

**Reply by ghost** [reply]
Ok I'll attempt to make simple example but I think that is hard as I never encountered that happening in my dev environment, maybe I'll try to replicate production with lot of user subscribing concurrently. 

>Should I create a cron/timer to verify whether 'pb.realtime.clientId' is empty and remove the user from the room?

For a temporary workaround can I do this?

**Reply by ganigeorgiev** [reply]
Again, I have no context about your use case and I'm not sure what to recommend.
In general, if running a timer and doing poll based checks works for your use case then I guess it should be fine.
But the best course of action would be to try to identify the main issue as it could hide a bigger/critical bug in your app or PocketBase.
